joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -63,12 +63,14 @@ trait AstCreatorHelper { this: AstCreator =>
     ast
   }
 
-  protected def fileLocalUniqueName(name: String, fullName: String, targetName: String = ""): (String, String) = {
+  protected def scopeLocalUniqueName(name: String, fullName: String, targetName: String = ""): (String, String) = {
```

Comment:
I'm wondering if this method could be made to automatically retrieve the fullname from the scope. I'm a little worried that the scope object and the provided full name might not always align, and we end up with clashing full names.

I think just for clarity it also wouldn't hurt to have multiple overloads of this method for the case where a fullname is needed or not (empty string).

Reply:
I changed it so we are now re-using the provided fullname (if any) to build the map key.
Only if the fullname is empty we fallback to the scope path.